### PR TITLE
Rework cudf::find_and_replace_all to use gather-based make_strings_column

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Reworks `cudf::find_and_replace_all` for strings to work with long strings and enable it to support large strings.
The custom kernels were replaced with a gather-based `make_strings_column` already optimized for long and short strings.
Large strings will automatically be supported in `make_strings_column` in a future PR.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by mhaseeb123 — APPROVED** [review]
Thank you for the contribution. Looks good to me!

File: cpp/src/strings/replace/find_replace.cu [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}  // namespace
+
+std::unique_ptr<cudf::column> find_and_replace_all(
+  cudf::strings_column_view const& input,
+  cudf::strings_column_view const& values_to_replace,
+  cudf::strings_column_view const& replacement_values,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  if (input.is_empty()) { return cudf::make_empty_column(type_id::STRING); }
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/src/strings/replace/find_replace.cu:70]: This is already handled by the caller:
https://github.com/rapidsai/cudf/pull/15305/files#diff-8c11e29e23c203bd58fd1ee8b6aca9d9b49e06edbaced8e68bbf4728f9a1ebc5R310-R312
So this check is probably not necessary. I can remove it.

**Review by pmattione-nvidia — APPROVED** [review]

**Review by KyleFromNVIDIA — APPROVED** [review]

**Review by bdice — APPROVED** [review]
CMake approval.

**Pull request merged**